gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java

```diff
@@ -966,6 +1038,81 @@ public SubscriptionEntity pause(final ExecutionContext executionContext, String
         }
     }
 
+    @Override
+    public SubscriptionEntity resumeByConsumer(final ExecutionContext executionContext, String subscriptionId) {
+        try {
+            logger.debug("Resume subscription by {} by consumer", subscriptionId);
+
+            Subscription subscription = subscriptionRepository
+                .findById(subscriptionId)
+                .orElseThrow(() -> new SubscriptionNotFoundException(subscriptionId));
+
+            final GenericPlanEntity genericPlanEntity = planSearchService.findById(executionContext, subscription.getPlan());
+            String apiId = genericPlanEntity.getApiId();
+            final GenericApiModel genericApiModel = apiTemplateService.findByIdForTemplates(executionContext, apiId);
+            validateConsumerStatus(subscription, genericApiModel);
+
+            if (subscription.getConsumerStatus() == Subscription.ConsumerStatus.PAUSED) {
+                Subscription previousSubscription = new Subscription(subscription);
+                final Date now = new Date();
+                subscription.setUpdatedAt(now);
+                subscription.setConsumerPausedAt(null);
+                subscription.setConsumerStatus(Subscription.ConsumerStatus.ACTIVE);
+
+                subscription = subscriptionRepository.update(subscription);
+
+                createAudit(
+                    executionContext,
+                    apiId,
+                    subscription.getApplication(),
+                    SUBSCRIPTION_RESUMED_BY_CONSUMER,
+                    subscription.getUpdatedAt(),
+                    previousSubscription,
+                    subscription
+                );
+
+                // active API Keys are automatically unpause
+                streamActiveApiKeys(executionContext, subscription.getId())
+                    .forEach(
+                        apiKey -> {
+                            apiKey.setPaused(false);
+                            apiKeyService.update(executionContext, apiKey);
+                        }
+                    );
```

Comment:
Duplicated from `restore` and `resume` methods.

Reply:
mutualized !